components/dashboards-web-component/src/designer/components/WidgetConfigurationPanel.jsx

```diff
@@ -0,0 +1,111 @@
+            this.wirePubSubInDashboardJSON(dashboardLayout.toConfig().content, selectedWidget.config.content[0].props.id,
+                event.currentTarget.id);
+            this.setState({checked: true});
+        } else {
+            if (pubsubComponent.unwire(selectedWidget.config.content[0].props.id, event.currentTarget.id)) {
+                this.unWirePubSubInDashboardJSON(dashboardLayout.toConfig().content,
+                    selectedWidget.config.content[0].props.id, event.currentTarget.id);
+                this.setState({checked: false});
+            }
+        }
+    }
+
+    wirePubSubInDashboardJSON(content, subscriberId, publisherId) {
+        content.forEach(contentItem => {
+            if (!(contentItem.type === 'component')) {
+                this.wirePubSubInDashboardJSON(contentItem.content, subscriberId, publisherId)
+            } else {
+                if (subscriberId === contentItem.props.id) {
+                    if (!contentItem.props.configs.pubsub.publishers) {
+                        contentItem.props.configs.pubsub.publishers = [];
+                    }
+                    contentItem.props.configs.pubsub.publishers.push(publisherId.toString())
+                }
+            }
+        });
+    }
+
+    unWirePubSubInDashboardJSON(content, subscriberId, publisherId) {
+        content.forEach(contentItem => {
+            if (!(contentItem.type === 'component')) {
+                this.unWirePubSubInDashboardJSON(contentItem.content, subscriberId, publisherId)
+            } else {
+                if (subscriberId === contentItem.props.id) {
+                    let position = contentItem.props.configs.pubsub.publishers.indexOf(publisherId);
+                    contentItem.props.configs.pubsub.publishers.splice(position, 1);
+                }
+            }
+        });
+    }
+
+    getPublishers() {
+        let publishers = [];
+        if (dashboardLayout.selectedItem &&
+            dashboardLayout.selectedItem.config.content[0].props.configs.pubsub.types.indexOf("subscriber") !== -1) {
+            for (let [key, val] of pubsubComponent.getPublishersMap().entries()) {
+                this.state.checked = !!(dashboardLayout.selectedItem.config.content[0].props.configs.pubsub.publishers &&
+                dashboardLayout.selectedItem.config.content[0].props.configs.pubsub.publishers.indexOf(val) !== -1);
+                publishers.push(<Checkbox id={val}
+                                          label={key.substring(0, key.length - 4)}
+                                          name={key}
+                                          onCheck={this.onCheck}
+                                          checked={this.state.checked}
+                                          className="publishers-list"/>)
+            }
+        }
+        return publishers;
+    }
+
+    render() {
+        return <Drawer open={this.props.open} openSecondary={true}
```

Comment:
Better to wrap this with parentheses (refer: https://github.com/airbnb/javascript/blob/master/react/README.md#parentheses)